### ISSUE TITLE
Building CCL fails with linking errors

### ISSUE THREAD

**Issue by anicola** [LSSTDESC/CCL#462]
Dear all,

I am trying to build CCL from the github repo. Following the instructions on the README runs, I run into these sort of errors

`    [  5%] Building CXX object CMakeFiles/libangpow.dir/src/walltimer.cc.o
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1454:no such instruction: `vmovsd 8(%rax), %xmm2'
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1457:no such instruction: `vmovsd 8(%r9), %xmm1'
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1458:no such instruction: `vcomisd %xmm1, %xmm2'
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1459:no such instruction: `vminsd %xmm2, %xmm1,%xmm1'
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1465:no such instruction: `vmovsd %xmm1, 8(%r8)'
    /var/folders/1b/387zdlq55fdg_mc9dc9mkr0w0000gn/T//cc09VLiw.s:1490:no such instruction: `vmovsd 8(%rdi,%rdx), %xmm1'`

I also run into the same errors when trying to install pyccl using pip.

To solve these problems, I removed all the -march=native commands in the Makefiles. This helps to some extent but I end up with 

`ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status`

I am running on MacOS 10.9 with gcc 4.9. 

Would you maybe have any idea what could be causing these problems?

Thank you very much for your help!
Best,
Andrina

**Comment by beckermr** [comment]
Well this bug is no fun. :( 

Which symbols are not found?

**Comment by anicola** [comment]
Hi Matt,

Thank you so much for coming back to me so quickly :-)!
As the whole error message is quite long, I've attached it as a txt file to this message. Please let me know if this helps or what other information I can provide.

Thanks so much again!
Best,
Andrina

[CCL-linking.txt](https://github.com/LSSTDESC/CCL/files/2419984/CCL-linking.txt)

**Comment by beckermr** [comment]
It appears the linker missed the C++ standard lib IIUIC. IDK how this happened OFC. Where di you get gcc 9? Homebrew? You may need the homebrew paths to the libs in the linking command somehow. 

If you can stomach it, you might try to reinstall without gcc 9 and use the bult-in clang instead. 

In general further work on the packaging here is needed! You should not have to deal with this!

**Comment by tilmantroester** [comment]
The linking issue might be related to the name of the C++ standard library. I think some compilers use `-lc++`, while others use `-lstdc++`. I think GCC uses the latter while clang uses the former, but this might be compiler version dependent and GCC 4.9 on OS X might be old enough to confuse some heuristic.

**Comment by beckermr** [comment]
Yeah. We really shouldn't be shipping our own compiler flags unless we plan to use an extensive configuration script. `conda` here we come...
